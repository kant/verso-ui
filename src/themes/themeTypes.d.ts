export interface ITheme {
    readonly colour: {
        primary: string;
        primaryHover: string;
        primaryActive: string;
        textOnPrimary: string;
        disabled: string;
        secondary: string;
        secondaryHover: string;
        secondaryActive: string;
        tertiary: string;
        tertiaryHover: string;
        tertiaryActive: string;
        accent: string;
        text: string;
        textInverted: string;
        formBackground: string;
    };
    readonly typography: {
        primaryFont: string;
        typeScale: ITypeScale;
        typeWeight: ITypeWeight;
    };
}
export interface ITypeScale {
    readonly heading1: string;
    readonly heading2: string;
    readonly heading3: string;
    readonly heading4: string;
    readonly large: string;
    readonly medium: string;
    readonly caption: string;
    readonly normal: string;
    readonly small: string;
}
export interface ITypeWeight {
    readonly light: number;
    readonly medium: number;
    readonly bold: number;
    readonly black: number;
}
