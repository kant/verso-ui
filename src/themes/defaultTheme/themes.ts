import { BLACK, NEUTRAL, RED, BLUE } from './colours'
import { primaryFont, typeScale, typeWeight } from './typography'
import { ITheme } from '../themeTypes'

export const defaultTheme: ITheme = {
  colour: {
    primary: BLACK[500],
    primaryHover: BLACK[400],
    primaryActive: BLACK[300],
    textOnPrimary: NEUTRAL[100],

    disabled: NEUTRAL[400],

    secondary: NEUTRAL[100],
    secondaryHover: NEUTRAL[200],
    secondaryActive: BLACK[300],

    tertiary: RED[200],
    tertiaryHover: RED[200],
    tertiaryActive: BLACK[300],

    accent: BLUE[200],

    text: BLACK[500],
    textInverted: NEUTRAL[100],

    formBackground: NEUTRAL[200]
  },
  typography: {
    primaryFont,
    typeScale,
    typeWeight
  }
}

export const darkTheme: ITheme = {
  colour: {
    primary: NEUTRAL[100],
    primaryHover: BLACK[400],
    primaryActive: BLACK[300],
    textOnPrimary: BLACK[500],

    disabled: NEUTRAL[400],

    secondary: BLACK[500],
    secondaryHover: NEUTRAL[200],
    secondaryActive: BLACK[300],

    tertiary: RED[200],
    tertiaryHover: RED[200],
    tertiaryActive: BLACK[300],

    accent: BLUE[200],

    text: NEUTRAL[100],
    textInverted: BLACK[500],

    formBackground: NEUTRAL[200]
  },
  typography: {
    primaryFont,
    typeScale,
    typeWeight
  }
}
