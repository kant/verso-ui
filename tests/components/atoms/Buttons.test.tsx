import React from 'react' // TODO:  TS warnings get produced as this is you don't need to import react anymore to use JSX, what can we do to mute this warning? https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports
import { jsx } from '@emotion/react'
import renderer from 'react-test-renderer'
import { PrimaryButton, SecondaryButton, TertiaryButton } from '../../../src'
import { defaultTheme } from '../../../src/themes'

test('Snapshot Test: Primary Button', () => {
  const component = renderer.create(
    <PrimaryButton
      isDisabled={false}
      label="Primary Button"
      theme={defaultTheme}
    />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Snapshot Test: Secondary Button', () => {
  const component = renderer.create(
    <SecondaryButton
      isDisabled={false}
      label="Secondary Button"
      theme={defaultTheme}
    />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Snapshot Test: Tertiary Button', () => {
  const component = renderer.create(
    <TertiaryButton
      isDisabled={false}
      label="Tertiary Button"
      theme={defaultTheme}
    />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
