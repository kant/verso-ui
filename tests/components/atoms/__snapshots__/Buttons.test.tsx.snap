// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test: Primary Button 1`] = `
<button
  className="css-1195o8y-PrimaryButton"
  type="button"
>
  Primary Button
</button>
`;

exports[`Snapshot Test: Secondary Button 1`] = `
<button
  className="css-1ou42su-SecondaryButton"
  type="button"
>
  Secondary Button
</button>
`;

exports[`Snapshot Test: Tertiary Button 1`] = `
<button
  className="css-11f3soy-TertiaryButton"
  type="button"
>
  Tertiary Button
</button>
`;
